lib/streamlit/elements/widgets/slider.py

```diff
@@ -618,6 +618,14 @@ def all_same_type(items):
             max_value = _datetime_to_micros(max_value)
             step = _delta_to_micros(cast(timedelta, step))
 
+        # The frontend will error if the values are equal, so checking here
+        # lets us produce a nicer python error message and stack trace.
+        if min_value == max_value:
+            raise StreamlitAPIException(
+                "Slider `min_value` must be less than the `max_value`."
+                f"The values were {min_value} and {max_value}."
```

Comment:
I might be slightly worried with the min_value and max_value being weird for the date formats. Might be better to exclude for simplicity. unless things look good enough.

Reply:
It looked bad, but moving the check to before we convert dates to ints made it reasonable.